### DISCUSSION TITLE
Compare string and float

### DISCUSSION THREAD

**Discussion by fabricioavil** [Mariusthvdb/custom-ui#121] [Q&A]
TLDR: Is there a way I can convert in customize.yaml the state type to compare two entities? Example, convert the state from string to float.

Hi,

I have a lot of customization in customize.yaml working perfectly, but the one on my temp sensors do not work correctly when the temperature goes below 10 degrees, it uses the "hot" color (red) instead of the "cold" color (blue).

Interestingly, it works correctly with "warm" (orange - above 25 degrees), "hot" (red - above 28 degrees), and "normal" (green - between 10 and 25 degrees), so it is only with the "cold" (below 10 degrees) that I get an unexpected color.

Config in customize.yaml:
```
sensor.temp_and_humidity_sensor_outdoor:
  friendly_name: Outdoor
  templates:
    icon_color: 'if (state >= entities[''input_number.climate_temperature_outdoor_hot_indicator''].state) return ''red''; else if (state >= entities[''input_number.climate_temperature_outdoor_warm_indicator''].state) return ''rgb(244, 180, 0)''; else if (state <= entities[''input_number.climate_temperature_outdoor_cold_indicator''].state) return ''RoyalBlue''; return ''rgb(4, 231, 98)'';'
```

Example, `sensor.temp_and_humidity_sensor_outdoor` has 6 degrees, but it gets `red` instead of `RoyalBlue`.

Trying to understand why, I did some tests in the Developer_Tools/Template.

1) When I compared without converting the type, it gives `False`, where it "should" be `True`:
{{ states('sensor.temp_and_humidity_sensor_outdoor') <= states('input_number.climate_temperature_outdoor_cold_indicator') }}

2) When I try to change one, any of them to float, using | float, I get a msg saying I can't compare `str` and `float`, so it means both are string:

```
{{ states('sensor.temp_and_humidity_sensor_outdoor') | float <= states('input_number.climate_temperature_outdoor_cold_indicator') }}

TypeError: '<=' not supported between instances of 'float' and 'str'
```

or

```
{{ states('sensor.temp_and_humidity_sensor_outdoor') <= states('input_number.climate_temperature_outdoor_cold_indicator') | float }}

TypeError: '<=' not supported between instances of 'str' and 'float'
```

3) If I convert both to float, it gives `True`, what should be what I need:

```
{{ states('sensor.temp_and_humidity_sensor_outdoor') | float <= states('input_number.climate_temperature_outdoor_cold_indicator') | float }}
```

So, it there a way I can convert in customize.yaml from str to float? If so, I will convert both, the sensor and the input_number) to float.

Thank you!

**Comment by Mariusthvdb** [comment]
this is more of a js question, than an issue with custom-ui, so please search the community or simply google.

you need ParseFloat() to do that

```
        icon: >
          return parseFloat(state) === 0 ? 'mdi:shield-check': 'mdi:alert';
        icon_color: >
          return parseFloat(state) === 0 ? 'green': 'red';
```

or:
```
        icon_color: >
          var vol = parseFloat(state);
          if (vol === 0.0) return 'gray';
          if (vol <= 0.3) return 'dodgerblue';
          if (vol <= 0.6) return 'green';
          if (vol <= 0.8) return 'orange';
          return 'firebrick';
```

closing, and happy to help in the community if still not working
